File: java/src/main/java/com/dokmatiq/docgen/model/TableData.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.util.List;

/** Table data with columns, rows, and optional styling. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record TableData(
        List<ColumnDef> columns,
        List<List<String>> rows,
        TableStyle style
) {
    public TableData(List<ColumnDef> columns, List<List<String>> rows) {
        this(columns, rows, null);
    }
}

File: php/src/Model/TableData.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** Table data with columns, rows, and optional styling. */
final class TableData
{
    /**
     * @param ColumnDef[] $columns
     * @param string[][] $rows
     */
    public function __construct(
        public readonly array $columns,
        public readonly array $rows,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return [
            'columns' => array_map(fn (ColumnDef $c) => $c->toArray(), $this->columns),
            'rows' => $this->rows,
        ];
    }
}
